.travis.yml

```diff
@@ -257,7 +257,15 @@ jobs:
         - wget https://scrutinizer-ci.com/ocular.phar
         - php ocular.phar code-coverage:upload --format=php-clover clover.xml
 
+    - stage: Pull request coding standard
+      if: type = pull_request
+      php: 7.2
+      script:
+        - wget https://github.com/diff-sniffer/git/releases/download/0.1.0/git-phpcs.phar
+        - php git-phpcs.phar origin/$TRAVIS_BRANCH...$TRAVIS_PULL_REQUEST_SHA
```

Comment:
Careful here: travis clone operations are generally shallow, which means that this may fail after 100 commits or even less.

Reply:
From the [documentation](https://docs.travis-ci.com/user/pull-requests/#How-Pull-Requests-are-Tested):

> Rather than test the commits that have been pushed to the branch the pull request is from, we test the merge between the origin and the upstream branch.

I was under the impression that both `origin/$TRAVIS_BRANCH` and `$TRAVIS_PULL_REQUEST_SHA` will be immediate parents of the merge commit. So they both should be on the second level of depth.

Not sure how shallow clone handles multiple parents though.

Reply:
[In theory](https://stackoverflow.com/questions/27059840/how-to-fetch-enough-commits-to-do-a-merge-in-a-shallow-clone), there indeed may be a problem when the merge base of the two parents is more than [50 commits](https://travis-ci.org/doctrine/dbal/jobs/317847162#L422) behind any of them. So even if we fetch the merge base manually, there's no guarantee all commits in between are available. Hm…